pkg/analyze/selector/iterator.go

```diff
@@ -0,0 +1,93 @@
+package selector
+
+import (
+	"github.com/scylladb/scylla-operator/pkg/analyze/selector/internal/matcher"
+	"github.com/scylladb/scylla-operator/pkg/analyze/selector/internal/predicate"
+	"github.com/scylladb/scylla-operator/pkg/analyze/selector/internal/spec"
+	"github.com/scylladb/scylla-operator/pkg/analyze/snapshot"
+)
+
+type Iterator struct {
+	spec   *spec.Spec
+	values map[string][]any
+}
+
+func (s *Selector) FromSnapshot(snapshot snapshot.Snapshot) (*Iterator, error) {
```

Comment:
```suggestion
func (s *Selector) IteratorFromSnapshot(snapshot snapshot.Snapshot) (*Iterator, error) {
```
and move to selector.go

Reply:
Done